kyo-prelude/shared/src/main/scala/kyo/Parse.scala

```diff
@@ -0,0 +1,473 @@
+package kyo
+
+import kyo.Ansi.*
+import kyo.kernel.*
+
+/** The Parse effect combines three fundamental capabilities needed for parsing:
+  *   - State management (Var[Parse.State]) to track input position
+  *   - Choice for handling alternatives and backtracking
+  *   - Error handling (Abort[ParseFailed]) for parse failures
+  *
+  * This combination enables building complex parsers that can handle ambiguous grammars, implement look-ahead, and provide detailed error
+  * messages.
+  */
+opaque type Parse <: (Var[Parse.State] & Choice & Abort[ParseFailed]) = Var[Parse.State] & Choice & Abort[ParseFailed]
+
+object Parse:
```

Comment:
The effect isn't very optimized yet. We could, for example, merge the features needed from `Choice`, `Var`, and `Abort` into a new `ArrowEffect` for better performance. I'm planning to follow up on benchmarking and optimization